GitUI/MessageBoxes.cs

```diff
@@ -56,6 +55,9 @@ internal MessageBoxes()
 
         private static MessageBoxes Instance => instance ?? (instance = new MessageBoxes());
 
+        public static bool Confirm([CanBeNull] IWin32Window owner, string text, string caption)
+            => Show(owner, text, caption, MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes;
+
```

Comment:
I'd very much prefer to remove this, update calling methods signatures and and update all call sites to verify against `DialogResult`. We already have too many convenience methods with insignificant value-add.
https://github.com/gitextensions/gitextensions/pull/8303 was inspired by all the same reasons.

I'd want a close look at each of these methods here, most of these methods have only a single call site, meaning we generate additional layers of code and consume cpu cycles for questionable reasons.

Reply:
> I'd very much prefer to remove this, update calling methods signatures and and update all call sites to verify against DialogResult. We already have too many convenience methods with insignificant value-add.
#8303 was inspired by all the same reasons.

It is not only about the comparison with `DialogResult`. All share the same common arguments. This is where the DRY principle *begins* to become applicable for me.
This is different from #8303, (which IMO bloats all callsites with arguments which are required for one or two special cases. I agree that there were too many overloaded ctors.)

Though we could declare `Confirm` `private` since there are no other users ATM.
Or remove it if you insist.

> I'd want a close look at each of these methods here, most of these methods have only a single call site, meaning we generate additional layers of code and consume cpu cycles for questionable reasons.

Don't concern yourself about this:
1. The compiler and linker can recognize and optimize this.
2. We are showing a MessageBox.

Reply:
> which IMO bloats all callsites with arguments which are required for one or two special cases.

Yes, it is a concern I share. But we have to "bloat" first to be able to see what and how API are used before we can architect a better solution.